### PR TITLE
Create Dropdown as a new UI element

### LINKED ISSUE #501: Create dropdown as a new UI element
### Overview

Create dropdown as a new UI element (based on finalized 0.4 Style Guide).

Find mockup here => [Finalized v0.4 Style Guide (Figma), page 2](https://www.figma.com/file/9c6pxjImN8MjHAZLH7zxYm/VRMS-Mocks?node-id=25453%3A28)

dropdown

### Action Items

- [x] Explore existing implementation for UI elements:
- `src/sass/elements`
- DEV UI KIT web page (run locally) http://localhost:3001/dev-ui-kit 
- DEV UI KIT component https://github.com/hackforla/VRMS/blob/development/client-mvp-04/src/utils/uiKit/uiKit.js

- [x]  Create a new section `/*** DROPDOWN ***/` section in [`src/utils/uiKit`], add JSX for new UI element inside this section (https://github.com/hackforla/VRMS/blob/development/client-mvp-04/src/utils/uiKit/uiKit.js)
- [x] Create a new scss file `_dropdown.scss` inside `src/sass/elements`
- [x] Add styles for new UI element inside `_dropdown.scss`
- [x] Define in the component data object that simulating JSON data coming from the API:
```
const data = {
   timezones : ["HST | Hawaii Standard Time", "AKST | Alaska Standard Time", "PST | Pacific Standard Time", "MST | Mountain Standard Time", "CST | Central Standard Time", "EST | Eastern Standard Time"]
 }
```
Display this data inside the dropdown
- [x]  When the user selects an option from the dropdown, save it to variable

### Resources/Instructions
[Finalized v0.4 Style Guide (Figma)](https://www.figma.com/file/9c6pxjImN8MjHAZLH7zxYm/VRMS-Mocks?node-id=25453%3A28)
Dev UI Kit web page (run locally) http://localhost:3001/dev-ui-kit

### PR DESCRIPTION
**Pull request by piall**
Fixes #501 

This can be implemented in many ways. I have found we can implement it without using lots of JS code or any JS plugins. If we use any predefined HTML element. Because already predefined event listeners and other things are added already. If we make it fully from scratch have to make those on our own. In this case, I have used `radio input` to get the job done by wrapping it with the `label` tag.

### CODE REVIEW

File: client-mvp-04/src/utils/uiKit/uiKit.js
Comment (drubgrubby) [inline client-mvp-04/src/utils/uiKit/uiKit.js]: I believe this is supposed to be a generic component to be reused by designers, so maybe the variable should be something generic, rather than specific to the TimeZone used for the demo.

File: client-mvp-04/src/utils/uiKit/uiKit.js
Comment (drubgrubby) [inline client-mvp-04/src/utils/uiKit/uiKit.js:32]: The original issue said, "save it to variable" rather than "save to state".  I'm wondering if one is preferable to the other in this instance. As I don't know the greater vision very well, hopefully Olga will be able to weigh in on this.

**Review by drubgrubby — COMMENTED** [review]
@mdPial -> Since @OlgaBilogurova  hasn't had time to review this yet, I'm leaving a couple of comments.  We will still need comments from her for a final decision on this as I'm not sure yet what the greater vision is, but these are some questions/thoughts I have.

1) In the image from issue #501 , it looks like there are a couple of letters typed in, and then the dropdown is showing some results.  Is this supposed to be a dropdown showing search results? 

https://github.com/hackforla/VRMS/issues/501#issue-825187235

If it is not showing search results, what should the initial state show?  It is currently showing the first item in the data file, should it instead show some text, like "Please make a selection", or should it be empty, or...? 

2) At the right edge of the drop-down image in Figma (and in the initial issue), there is an arrow pointing down.  Is that arrow supposed to rotate depending on if the drop-down is displaying results? 

Again, these are comments and I'm not sure if you should make these changes before Olga comments.  Or perhaps we can sort it out at the meeting tonight.

File: client-mvp-04/src/utils/uiKit/uiKit.js
Comment (piall) [reply client-mvp-04/src/utils/uiKit/uiKit.js:32]: Yes. I thought maybe indirectly meant to save it in the state for the future possibility.

**Comment by piall** [comment]
> In the image from issue #501 , it looks like there are a couple of letters typed in, and then the dropdown is showing some results. Is this supposed to be a dropdown showing search results?

I don't think it's showing search results. What I have understood the saved data in variable shows there. As the mockup data is just an example, Ms. Olga will be able to give better clarification.

> If it is not showing search results, what should the initial state show? It is currently showing the first item in the data file, should it instead show some text, like "Please make a selection", or should it be empty, or...?

I am also confused about that. As Ms. Olga was very busy at that time I have not disturbed her. I thought I will make changes if needs after making PR and when she will have time to review.

File: client-mvp-04/src/sass/elements/_dropdown.scss
Comment (OlgaBilogurova) [inline client-mvp-04/src/sass/elements/_dropdown.scss]: Also, to demonstrate how select works without collapsing section let's add `height: 380px;` and remove these 3 lines.

File: client-mvp-04/src/sass/elements/_dropdown.scss
Comment (OlgaBilogurova) [inline client-mvp-04/src/sass/elements/_dropdown.scss]: Class `.dropdown-container` is related just to `DevUiKit`. 
Inside the `Dropdown `component, we don't have this wrapper.
So, you need to move these styles to `uiKit.scss.`

File: client-mvp-04/src/sass/elements/_dropdown.scss
Comment (OlgaBilogurova) [inline client-mvp-04/src/sass/elements/_dropdown.scss]: Now when select opens options it's moving a little bit.
I think it's because you have width: 230px for `dropdown `and for `option-container` 232px.
It needs to be consistent => 230px.

File: client-mvp-04/src/sass/elements/_dropdown.scss
Comment (OlgaBilogurova) [inline client-mvp-04/src/sass/elements/_dropdown.scss]: Please, replace with `margin: 0 0 6px 3px;`
When setting `0`, you don't need `px`.

File: client-mvp-04/src/sass/elements/_dropdown.scss
Comment (OlgaBilogurova) [inline client-mvp-04/src/sass/elements/_dropdown.scss]: Replace with `margin: auto 14.8px 12px 0;`
The same as above.

File: client-mvp-04/src/sass/elements/_dropdown.scss
Comment (OlgaBilogurova) [inline client-mvp-04/src/sass/elements/_dropdown.scss]: Please, remove` !important` here.

**Review by OlgaBilogurova — CHANGES_REQUESTED** [review]
Hi @mdPial @drubgrubby,

Just started review this PR. Here are a few comments regarding styles. 
@mdPial please, wait for the full review. I hope I'll finish it tomorrow.

Regarding the `search `and` animation for the arrow`. I suppose we don't need this right now.
Since we need to implement more important functionality for `mvp-04`.

File: client-mvp-04/src/components/common/dropdown/dropdown.js
Comment (OlgaBilogurova) [inline client-mvp-04/src/components/common/dropdown/dropdown.js]: When using `className value` as a `string` it's enough to put here a `string` without curly brackets: `className='dropdown'`. In case when you're passing `props `or use `condition statements` you need them for sure.

File: client-mvp-04/src/components/common/dropdown/dropdown.js
Comment (OlgaBilogurova) [inline client-mvp-04/src/components/common/dropdown/dropdown.js]: Replace with `<div className={'arrow'} />`

File: client-mvp-04/src/utils/uiKit/uiKit.js
Comment (OlgaBilogurova) [reply client-mvp-04/src/utils/uiKit/uiKit.js:32]: Yeah, `save to the state` it's the even better solution.
As well, as creating a separate reusable `Dropdown `component. Good job @mdPial.

File: client-mvp-04/src/utils/uiKit/uiKit.js
Comment (OlgaBilogurova) [reply client-mvp-04/src/utils/uiKit/uiKit.js]: Agree with this. `Dropdown `is a reusable generic component but to be reused by developers (not designers).
Let's rename `setTimezone `to `setSelectedOption`.
`<Dropdown data={data} setSelectedOption={setTimezone} />`

File: client-mvp-04/src/components/common/dropdown/dropdown.js
Comment (OlgaBilogurova) [inline client-mvp-04/src/components/common/dropdown/dropdown.js]: Replace with:
`const Dropdown = ({ data, setSelectedOption }) => {`

File: client-mvp-04/src/components/common/dropdown/dropdown.js
Comment (OlgaBilogurova) [inline client-mvp-04/src/components/common/dropdown/dropdown.js]: Replace with:
`setSelectedOption(event.target.value);`

File: client-mvp-04/src/components/common/dropdown/dropdown.js
Comment (OlgaBilogurova) [inline client-mvp-04/src/components/common/dropdown/dropdown.js]: As you don't have `class` for `clickStatus`.
This line could be replaced with:
`<div className={`option-container ${!clickStatus && 'hide'}`}>`

**Review by OlgaBilogurova — CHANGES_REQUESTED** [review]
Good job @mdPial! 👍
It's great to have a new reusable `Dropdown `component for `mvp-04`.
This is my full review. Please, take a look at my suggestions.

File: client-mvp-04/src/components/common/dropdown/dropdown.js
Comment (piall) [reply client-mvp-04/src/components/common/dropdown/dropdown.js]: I was following previous coding styles. I was confused with that. I thought it maybe for good code quality 😂️

File: client-mvp-04/src/sass/elements/_dropdown.scss
Comment (mdPial) [reply client-mvp-04/src/sass/elements/_dropdown.scss:54]: image
I hade used `!important` to solve this issue. Because tags render and take spaces in the UI.

**Comment by piall** [comment]
> Good job @mdPial! 

Thank you @OlgaBilogurova . :smile: 

 Sorry, for the late response. I was preparing & celebrating the festival.

File: client-mvp-04/src/sass/elements/_dropdown.scss
Comment (piall) [reply client-mvp-04/src/sass/elements/_dropdown.scss]: Actually, in the Figma UI it was like this. Options container is `232`px and options are `230`px. If we make equal then this happens. 
image

Comment (piall) [reply client-mvp-04/src/sass/elements/_dropdown.scss]: > Actually, in the Figma UI it was like this. Options container is `232`px and options are `230`px. If we make equal then this happens.

I resolved it by making **option-label** `width : 100%`

File: client-mvp-04/src/sass/elements/_dropdown.scss
Comment (mdPial) [reply client-mvp-04/src/sass/elements/_dropdown.scss:54]: I have resolved it.

File: client-mvp-04/src/components/common/dropdown/dropdown.js
Comment (piall) [reply client-mvp-04/src/components/common/dropdown/dropdown.js]: Why we are following two different style?. In uiKit.js we are using `{}` for all class

Comment (OlgaBilogurova) [reply client-mvp-04/src/components/common/dropdown/dropdown.js]: Yes, we need to replace {} with " " inside the uiKit as well. 
If you want you can create a separate PR for this refactoring.

**Review by OlgaBilogurova — APPROVED** [review]

File: client-mvp-04/src/components/common/dropdown/dropdown.js
Comment (mdPial) [reply client-mvp-04/src/components/common/dropdown/dropdown.js]: Okay. I will create a separate PR for that refactoring. Should I create an issue for that first?

**Pull request merged**
